### PR TITLE
Fix ambiguity about custom providers in containers.adoc

### LINKED ISSUE #22161: Resolve ambiguity in documentation re: building a Docker image with custom provider(s)
### Description

See discussion [Confusing doc: building Docker image with custom provider(s)](https://github.com/keycloak/keycloak/discussions/22004). The problem is that [Server Guide for Running Keycloak in a container](https://github.com/keycloak/keycloak/blob/main/docs/guides/server/containers.adoc) provides the mechanism for adding a custom provider to a Docker image, but does not clearly state _where_ in the Dockerfile to place the provided line of code.

Collaborator [vmusikar](https://github.com/vmuzikar) clarified that the custom provider(s) must be ADDed to the image prior to running `build`.

The task here is to make the documentation clear on this point.

As I stated in the discussion, I intend to submit a pull request to resolve this issue.

### PR DESCRIPTION
**Pull request by erickhagstrom**
Amplified the explanation of how to include custom providers in a Dockerfile.

Added code to provide context.

Closes #22161

### CODE REVIEW

File: docs/guides/server/containers.adoc [hunk 1]
```diff
@@ -54,12 +54,24 @@ The build process includes multiple stages:
 * In the final image, additional configuration options for the hostname and database are set so that you don't need to set them again when running the container.
 * In the entrypoint, the `kc.sh` enables access to all the distribution sub-commands.
 
-To install custom providers, you just need to define a step to include the JAR file(s) into the `/opt/keycloak/providers` directory:
+To install custom providers, you just need to define a step to include the JAR file(s) into the `/opt/keycloak/providers` directory.
+This step must be placed _after_ the first `FROM` line and before the line that `RUNs` the `build` command, as below:
```
Comment (vmuzikar) [inline docs/guides/server/containers.adoc:58]: I would not emphasise the first `FROM`. There can be some more stages, see e.g.
https://github.com/keycloak/keycloak/blob/816ed5d3e0f0885a6bb57034be2433a7c7ac78aa/docs/guides/server/containers.adoc?plain=1#L79-L84

File: docs/guides/server/containers.adoc [hunk 2]
```diff
@@ -54,12 +54,24 @@ The build process includes multiple stages:
...
 * In the entrypoint, the `kc.sh` enables access to all the distribution sub-commands.
 
-To install custom providers, you just need to define a step to include the JAR file(s) into the `/opt/keycloak/providers` directory:
+To install custom providers, you just need to define a step to include the JAR file(s) into the `/opt/keycloak/providers` directory.
+This step must be placed _after_ the first `FROM` line and before the line that `RUNs` the `build` command, as below: 
 
 [source, dockerfile]
 ----
 # A example build step that downloads a JAR file from a URL and adds it to the providers directory
+# Context: the first `FROM` line
```
Comment (vmuzikar) [inline docs/guides/server/containers.adoc:63]: Same as above.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Thank you for the contribution!

File: docs/guides/server/containers.adoc (hunk 1 above)
Comment (erickhagstrom) [reply docs/guides/server/containers.adoc]: Thank you. I have incorporated your suggestions.

File: docs/guides/server/containers.adoc (hunk 2 above)
Comment (erickhagstrom) [reply docs/guides/server/containers.adoc]: Thank you. I have incorporated your suggestions.

**Review by vmuzikar — APPROVED** [review]
@erickhagstrom Thanks, LGTM.

**Pull request merged**